File: mall-goods-interface/src/main/java/com/mall/goods/service/SpecificationOptionService.java
package com.mall.goods.service;

import com.mall.pojo.SpecificationOption;
import common.pojo.PageResult;

import java.util.List;

/**
 * 规格选项明细服务接口
 * @author Wwl
 *
 */
public interface SpecificationOptionService {

	/**
	 * 增加
	 * @param specificationOption
	 */
	public void add(SpecificationOption specificationOption);

	/**
	 * 修改
	 * @param specificationOption
	 */
	public void update(SpecificationOption specificationOption);

	/**
	 * 根据Id获取实体
	 * @param id
	 * */
	public SpecificationOption findOne(Long id);

	/**
	 * 查询所有
	 * @return
	 */
	public List<SpecificationOption> findAll();

	/**
	 * 查询所有-分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 根据条件查询，分页
	 * @param specificationOption
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(SpecificationOption specificationOption, int pageNum, int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(Long[] ids);

}

File: mall-manager-web/src/main/webapp/js/service/specificationOptionService.js
/**
 * 规格选项服务层
 * Created by Wwl on 2019/3/14
 */
app.service('specificationOptionService',function ($http) {

    //增加
    this.add = function (entity) {
        return $http.post('../specificationOption/add.do',entity);
    }

    //修改
    this.update = function (entity) {
        return $http.post('../specificationOption/update.do',entity);
    }

    //删除
    this.delete = function (ids) {
        return $http.get('../specificationOption/delete.do?ids='+ids);
    }

    //查询实体
    this.findOne = function (id) {
        return $http.get('../specificationOption/findOne.do?id='+id);
    }

    //查询列表
    this.findList = function(){
        return $http.get('../specificationOption/list.do');
    }

    //分页查询
    this.findPage = function(page, size){
        return $http.get('../specificationOption/listByPage.do?page=' + page + '&size=' + size);
    }

    //搜索
    this.search = function (page,size,searchEntity) {
        return $http.post('../specificationOption/search.do?page='+page+'&size='+size,searchEntity);
    }

});